saya/core/src/data_availability/celestia.rs

```diff
@@ -0,0 +1,183 @@
+use anyhow::Result;
+use celestia_rpc::{BlobClient, Client};
+use celestia_types::{nmt::Namespace, AppVersion, Blob, TxConfig};
+use log::{debug, info};
+use tokio::sync::mpsc::{Receiver, Sender};
+use url::Url;
+
+use crate::{
+    data_availability::{
+        DataAvailabilityBackend, DataAvailabilityBackendBuilder, DataAvailabilityCursor,
+        DataAvailabilityPacket, DataAvailabilityPayload, DataAvailabilityPointer,
+    },
+    service::{Daemon, FinishHandle, ShutdownHandle},
+};
+
+// TODO: make namespace configurable?
```

Comment:
In the current setup, there is one saya for one katana. Hence it will be important to configure it from the CLI (at least for now). Or we can update the `DataAvailabilityCursor` to contain the `chain_id` for less configuration at the high level.

In the long run, we will tend to have one saya that can aggregate jobs instead of having 1-1 matching. In this situation, a const would suffice.